generator/discovery/update_discovery_doc.sh

```diff
@@ -19,6 +19,26 @@ set -euo pipefail
 source "$(dirname "$0")/../../ci/lib/init.sh"
 source module ci/lib/io.sh
 
+function print_service_textproto() {
+  service_proto_path=$(echo "${1}" | sed -En 's/protos\/(google\/cloud\/compute\/.*\/v[[:digit:]]\/.*\.proto)/\1/p')
+  product_path=$(echo "${1}" | sed -En 's/protos\/(google\/cloud\/compute\/.*\/v[[:digit:]])\/.*\.proto/\1/p')
+  initial_copyright_year=$(date +"%Y")
+  echo "  rest_services {"
+  echo "    service_proto_path: \"${service_proto_path}\""
+  echo "    product_path: \"${product_path}\""
+  echo "    initial_copyright_year: \"${initial_copyright_year}\""
+  echo "    retryable_status_codes: [\"kUnavailable\"]"
+  echo "    generate_rest_transport: true"
+  echo "    generate_grpc_transport: false"
+  echo "  }"
+}
+
+function add_service_directory() {
+  service_dir=$(echo "${1}" | sed -En 's/protos\/google\/cloud\/compute\/(.*\/v[[:digit:]]\/).*\.proto/\1/p')
+  echo "    \"${service_dir}\""
+  sed -i "19i\    \"${service_dir}\"" "${PROJECT_ROOT}/google/cloud/compute/service_dirs.cmake"
```

Comment:
Maybe a comment explaining how the "19" skips the boilerplate and inserts this new line in the middle of the list? Also, it may be less error prone if you have `sed` search for the `# cmake-format: sort` comment?